tests/test_forward_link.py

```diff
@@ -0,0 +1,338 @@
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal, assert_series_equal
+
+from src.forward_link import get_link, zerofy_values
+
+
+class TestFilters:
+    # based on 02_C_FI_input
+    df = pd.DataFrame(
+        data={
+            "identifier": [20001, 20001, 20002, 20002, 20003, 20003, 20004, 20004],
+            "date": [202001, 202002, 202001, 202002, 202001, 202002, 202001, 202002],
+            "group": [100, 100, 100, 100, 100, 100, 100, 100],
+            "question": [2536.0, 8283.0, 9113.0, 2970.0, 5644.0, 989.0, np.nan, np.nan],
+            "other": [35, 35, 72, 72, 77, 77, 30, 30],
+        }
+    )
+
+    def test_basic_filter(self):
+        """Test a basic filter, filters questions with identifier different to 20001"""
+
+        expected = pd.DataFrame(
```

Comment:
It looks like the bulk of this dataframe is used a few times in the unit tests. I agree that having fewer files to load makes the repo cleaner, but having one csv which has the setup would remove the repeated setup. I also appreciate that the "question" column changes depending on the filter applied which makes it more difficult to use a csv

Reply:
I agree with you about this, another point is according to pytest docs

"Something to be aware of when grouping tests inside classes is that each test has a unique instance of the class. Having each test share the same class instance would be very detrimental to test isolation and would promote poor test practices"

Not sure about appropriate path to save it though, perhaps we should save them in a folder test_data ?